src/pages/LogOutPreviousUserPage.js

```diff
@@ -17,37 +24,43 @@ const propTypes = {
-            Linking.getInitialURL().then((transitionURL) => {
-                const sessionEmail = props.session.email;
-                const isLoggingInAsNewUser = SessionUtils.isLoggingInAsNewUser(transitionURL, sessionEmail);
-
-                if (isLoggingInAsNewUser) {
-                    Session.signOutAndRedirectToSignIn();
-                }
-
-                // We need to signin and fetch a new authToken, if a user was already authenticated in NewDot, and was redirected to OldDot
-                // and their authToken stored in Onyx becomes invalid.
-                // This workflow is triggered while setting up VBBA. User is redirected from NewDot to OldDot to set up 2FA, and then redirected back to NewDot
-                // On Enabling 2FA, authToken stored in Onyx becomes expired and hence we need to fetch new authToken
-                const shouldForceLogin = lodashGet(props, 'route.params.shouldForceLogin', '') === 'true';
-                if (shouldForceLogin) {
-                    const email = lodashGet(props, 'route.params.email', '');
-                    const shortLivedAuthToken = lodashGet(props, 'route.params.shortLivedAuthToken', '');
-                    Session.signInWithShortLivedAuthToken(email, shortLivedAuthToken);
-                }
-            });
-        },
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-        [],
-    );
+    useEffect(() => {
+        Linking.getInitialURL().then((transitionURL) => {
+            const sessionEmail = props.session.email;
+            const isLoggingInAsNewUser = SessionUtils.isLoggingInAsNewUser(transitionURL, sessionEmail);
+
+            if (isLoggingInAsNewUser) {
+                Session.signOutAndRedirectToSignIn();
+            }
+
+            // We need to signin and fetch a new authToken, if a user was already authenticated in NewDot, and was redirected to OldDot
+            // and their authToken stored in Onyx becomes invalid.
+            // This workflow is triggered while setting up VBBA. User is redirected from NewDot to OldDot to set up 2FA, and then redirected back to NewDot
+            // On Enabling 2FA, authToken stored in Onyx becomes expired and hence we need to fetch new authToken
+            const shouldForceLogin = lodashGet(props, 'route.params.shouldForceLogin', '') === 'true';
+            if (shouldForceLogin) {
+                const email = lodashGet(props, 'route.params.email', '');
+                const shortLivedAuthToken = lodashGet(props, 'route.params.shortLivedAuthToken', '');
+                Session.signInWithShortLivedAuthToken(email, shortLivedAuthToken);
+            }
+
+            const exitTo = lodashGet(props, 'route.params.exitTo', '');
+            if (exitTo && !props.account.isLoading && !isLoggingInAsNewUser) {
+                Navigation.isNavigationReady().then(() => {
+                    Navigation.navigate(exitTo);
+                });
+            }
+        });
+    }, [props]);
```

Comment:
Not really confirmed, but this _might_ be the root cause of https://github.com/Expensify/App/issues/33413. Let's try to be a little more careful with effect dependencies to avoid many unnecessary API calls 🙏